cmd/merge.go

```diff
@@ -94,7 +94,12 @@ func mergeOneRepo(r initialize.Repo, ctx context.Context) error {
 		RequireReviewApproval: !mergeFlagIgnoreReviewApproval,
 		RequireBuildSuccess:   !mergeFlagIgnoreBuildStatus,
 	}
-	output, err := merge.Merge(ctx, input, githubLimiter, mergeThrottle)
+	var output merge.Output
+	if r.Provider == "gitlab" {
+		output, err = merge.GitlabMerge(input, githubLimiter, mergeThrottle)
+	} else if r.Provider == "github" {
+		output, err = merge.GitHubMerge(ctx, input, githubLimiter, mergeThrottle)
+	}
```

Comment:
we should error in the case that it's neither of these (same in other if / else if blocks where we check `r.Provider`)

Reply:
I've checking provider here

```		if provider != "github" && provider != "gitlab" {
			log.Fatal("--provider must be github or gitlab")
		}```

and write provider in the state file, in order to not provide provider on future steps as plan, push, merge. But if you want to double-check it, I could provide another condition step